processFiles/initEpub.js

```diff
@@ -0,0 +1,58 @@
+const JSZip = require('jszip')
+const { parseOPF } = require('./parseOPF')
+
+exports.initEpub = async file => {
+  // Should check for src prop and download epub if file is not present.
+  // Should also check for activity prop in case we have an id for annotations
+  // Need to figure out a way to handle three scenarios: import file, import URL, read from url cached in file: file, src, activity. Set during init. initAsync only called during import preview or during book read.
+  // Book object only created during import or read. Listings are still activities.
+  //  const { file, zip } = context
+  const zip = await JSZip.loadAsync(file, {
+    base64: false
+  })
+
+  const container = await zip.file('META-INF/container.xml').async('string')
+  const result = container.match(/full-path="([^"]+)"/)
+
+  // We save the full path to the opf
+  const opfPath = result[1]
+  const opf = await zip.file(opfPath).async('string')
+  const book = parseOPF(opf, opfPath)
+  const media = book.resources.map(resource => {
+    return {
+      documentPath: resource.url,
+      mediaType: resource.encodingFormat,
+      json: {}
+    }
+  })
+  book.resources = book.resources.concat(
+    {
+      type: 'LinkedResource',
+      rel: ['alternate'],
+      url: 'original.epub',
+      encodingFormat: 'application/epub+zip'
+    },
+    {
+      type: 'LinkedResource',
+      rel: ['alternate'],
+      url: 'original.opf',
```

Comment:
I'm being a teeny-tiny nitpicky here, but ccould you replace `'original.opf'` here with a reference to the `opfPath` variable? Otherwise attempts to fetch the original will fail. (Yeah, I know. This bug was in my original. 🙂 )

So, something like ` url: opfPath`?

Other than that, looks great.